src/main/scala/org/zalando/kanadi/api/Subscriptions.scala

```diff
@@ -524,13 +542,197 @@ final case class CancelledByClient(subscriptionId: SubscriptionId, streamId: Str
+                                   oAuth2TokenProvider,
+                                   AvroSchema(eventTypeName, etSchema.schema.asString.get, etSchema.version.get))))
+
+    Await.result(result, Duration.apply(5, TimeUnit.SECONDS)).getOrElse(throw SchemaNotFoundError(consumerSchema))
+  }
+}
+class AvroSubscriptions[T](baseUri: URI,
+                           oAuth2TokenProvider: Option[OAuth2TokenProvider] = None,
+                           consumerSchema: AvroSchema)(implicit
+    kanadiHttpConfig: HttpConfig,
+    http: HttpExt,
+    materializer: Materializer,
+    tag: ClassTag[T])
+    extends Subscriptions(baseUri, oAuth2TokenProvider) {
+
+  import com.fasterxml.jackson.dataformat.avro.{AvroSchema => JacksonSchema}
+
+  private val userSchema = new JacksonSchema(consumerSchema.parsedSchema)
+  private val reader     = AvroUtil.AvroMapper.reader(userSchema)
+
+  def eventsStreamedSourceAvro(subscriptionId: SubscriptionId,
+                               connectionClosedCallback: Subscriptions.ConnectionClosedCallback =
+                                 Subscriptions.ConnectionClosedCallback { _ =>
+                                   ()
+                                 },
+                               streamConfig: Subscriptions.StreamConfig = Subscriptions.StreamConfig())(implicit
+      flowId: FlowId,
+      executionContext: ExecutionContext,
+      eventStreamSupervisionDecider: Subscriptions.EventStreamSupervisionDecider)
+      : Future[Subscriptions.NakadiSource[T]] = {
+    import akka.http.scaladsl.model.headers.Accept
+    val uri           = getStreamUri(subscriptionId, streamConfig)
+    val streamHeaders = Accept(MediaType.applicationBinary("avro-binary", Compressible)) +: getBaseHeaders
+
+    def cleanup(streamId: StreamId, cancelledByClient: Boolean) = {
+      // Cleaning up the connection afterwards
+      logger.info(s"SubscriptionId: ${subscriptionId.id}, StreamId: ${streamId.id} HTTP connection closed, cleaning up")
+      val connectionClosedData = Subscriptions.ConnectionClosedData(
+        OffsetDateTime.now(),
+        subscriptionId,
+        streamId,
+        cancelledByClient
+      )
+      connectionClosedCallback.connectionClosedCallback(connectionClosedData)
+      try killSwitches.remove((subscriptionId, streamId))
+      catch {
+        case NonFatal(e) =>
+          logger.warn(
+            s"SubscriptionId: ${subscriptionId.id}, StreamId: ${streamId.id}, error removing HTTP connection from pool",
+            e)
+      }
+    }
+
+    for {
+      headers <- oAuth2TokenProvider match {
+                   case None => Future.successful(streamHeaders)
+                   case Some(futureProvider) =>
+                     futureProvider.value().map { oAuth2Token =>
+                       toHeader(oAuth2Token) +: streamHeaders
+                     }
+                 }
+
+      request = HttpRequest(HttpMethods.GET, uri, headers)
+      _       = logger.debug(request.toString)
```

Comment:
it might be worth to be more descriptive in logging, if we want to add it, instead of only `request.toString`, to have more description in the log string.